libs/execution_context/ebpf_core.c

```diff
@@ -2424,12 +2418,9 @@ typedef struct _ebpf_protocol_handler
 } const ebpf_protocol_handler_t;
 
 #define PROTOCOL_NATIVE_MODE 1
-// #if !defined(CONFIG_BPF_JIT_DISABLED)
```

Comment:
I understand all the changes before this line, which are fine.
But I don't understand why 2427-2423 are changed.
Can you either explain or revert these lines?

Reply:
So there are 2 parts for the changes in this section.

Part 1: The protocol handlers that are defined today use `PROTOCOL_*_MODE` flags to define in what modes can they be called. Since we are uncommenting JIT IOCTLs for native mode also, we need to at least define these macros (looks like `PROTOCOL_INTERPRET_MODE` does not really need to be defined, and I have disabled that back for native mode).

Part 2: We have 2 flags `CONFIG_BPF_JIT_DISABLED` and `CONFIG_BPF_INTERPRETER_DISABLED` which theoretically creates 4 possible combinations. Currently we were only handling 3 combinations (case of `!defined(CONFIG_BPF_INTERPRETER_DISABLED)` was not being handled). So this PR adds that case also.

There was another bug I think where we are using `||` instead of `&&` when we are defining `PROTOCOL_JIT_OR_INTERPRET_MODE` to be `(PROTOCOL_JIT_MODE | PROTOCOL_INTERPRET_MODE)`